### ISSUE TITLE
fix to #6734 breaks pipe with multi-instanciated modules

### ISSUE THREAD

**Issue by chrisaga** [darktable-org/darktable#7565]
**Describe the bug**

The issue #6734 has been fixed in darktable 3.4 but the fix introduced a new issue when there are multiple instances of the same module in the development pipe. The photos is more or less scrambled and you can't access anymore to the extra instances of the multi-instanced module.

N.B. **If there is no multi-instanciated module in the pipe everything seems to be OK**

**To Reproduce**
Same thing as for #6734 except the result is different.

1. Edit a photo with for instance two instances of the exposure module
1. Go to 'Light Table'
2. Select the photo
3. Apply a style (I tested with a style which adds a watermark)
4. See that the thumbnail is correctly updated
5. Press Ctrl-Z to "undo"
6. See that the thumbnail is incorrectly updated (it is more or less scrambled, sometimes looking a little bit like a raw under/overexposure warning, sometimes looking like heavy noise).
7. Go to 'Dark Room'
8. See that there is a lot of "garbage" as in the thumbnail
9. See that the second instance of the exposure module is no more in the pipe, but your history stack is not affected
10. See that Darktable throws this kind of error message to stderr :
```
cannot get iop-order for exposure instance 1
[dt_ioppr_check_iop_order] history module not used but enabled!! exposure 1(2147483647) image 8747 (dt_dev_read_history_no_image end)
```

**Expected behavior**
No garbage
Second instance of the exposure module kept in the pipe

**Screenshots**

![Screenshot_20201230_172401](https://user-images.githubusercontent.com/31566103/103366902-0271f400-4ac4-11eb-95c5-9e1514c6fbde.png)

**Platform (please complete the following information):**

*    Darktable Version: [3.4.0]
*    OS: [ Ubuntu 20.04 with KDE 5 ]
*    OpenCL not activated
*    Graphics card : Cape Verde XT [Radeon HD 7770/8760 / R7 250X]

**Additional context**
I experimented it with two instances of exposure module and with two instances of color balance module.
The xmp sidecar file seems to be OK but removing the photo from the database and re-importing it with its xmp file gets the same garbage.
The only way to partially recover your work is to go back in the history stack before the first duplicate instance of whatever module is duplicated. Or if you have important settings after that to manually edit your xmp file and remove what is related to multiple instances of modules (keep only the first instance). Then re-import it

**Comment by TurboGit** [comment]
Do you confirm that you apply the style in overwrite mode ?

**Comment by chrisaga** [comment]
> Do you confirm that you apply the style in overwrite mode ?

Yes, I confirm. It really seems to be related to the fix.

**Comment by TurboGit** [comment]
Should be fixed in #7572. Do you compile dt yourself ? If so can you test this PR before it gets merged. Let me know.

**Comment by chrisaga** [comment]
> Should be fixed in #7572. Do you compile dt yourself ? If so can you test this PR before it gets merged. Let me know.

Unfortunately, I don't compile dt myself. Maybe I will if I get more involved in the project. It's not a priority for me now since I have other IT stuffs to attend too and a business to start.
If you feel it's really important, I may give a try next week.

**Comment by TurboGit** [comment]
@chrisaga : No, thanks. I'll merge and you'll be able to use a nightly package to test.
